### ISSUE TITLE
[🐛BUG] RuntimeError: gather_out_cuda(): Expected dtype int64 for index

### ISSUE THREAD

**Issue by johnny12150** [RUCAIBox/RecBole#764]
**Describe the bug**
Runtime error due to the type of tensor is `int32` instead of `int64`.
This happens on SASRec, NARM,TransRec.
```
  File "F:/volume/SR baseline/run_recbole.py", line 23, in <module>
    run_recbole(model=args.model, dataset=args.dataset, config_file_list=config_file_list, config_dict=arg_dict)
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\quick_start\quick_start.py", line 54, in run_recbole
    train_data, valid_data, saved=saved, show_progress=config['show_progress']
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\trainer\trainer.py", line 262, in fit
    train_loss = self._train_epoch(train_data, epoch_idx, show_progress=show_progress)
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\trainer\trainer.py", line 152, in _train_epoch
    losses = loss_func(interaction)
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\model\sequential_recommender\sasrec.py", line 127, in calculate_loss
    seq_output = self.forward(item_seq, item_seq_len)
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\model\sequential_recommender\sasrec.py", line 121, in forward
    output = self.gather_indexes(output, item_seq_len - 1)
  File "C:\Users\Wade\Anaconda3\envs\rosetta_GNN\lib\site-packages\recbole\model\abstract_recommender.py", line 119, in gather_indexes
    output_tensor = output.gather(dim=1, index=gather_index)
RuntimeError: gather_out_cuda(): Expected dtype int64 for index
```

**To Reproduce**
1. extra yaml file
I use Tmall-buy as an example.
```
USER_ID_FIELD: user_id
load_col:
  inter: [user_id, item_id, timestamp]
epochs: 30
train_batch_size: 100
eval_batch_size: 100  # val and test batch_size
topk: [10, 20]
valid_metric: MRR@20
stopping_step: 5  # early stop after num epochs
split_ratio: [0.8,0.1,0.1]
```
2. Use recoble_run() to train.

**Expected behavior**
Models should be able to train.

**Screenshots**
![image](https://user-images.githubusercontent.com/22369387/110666942-b9959480-8204-11eb-996e-50df228e553a.png)

**Desktop (please complete the following information):**
 - OS: Windows
- RecBole Version 0.2.0
 - Python Version 3.7 & 3.8
- PyTorch Version 1.7.1 & 1.8.0
- cudatoolkit Version 10.2 & 11

I have encountered this n Linux with `conda` install PyTorch 1.8.0.

**Comment by 2017pxy** [comment]
@johnny12150 Hi, we have already fixed this bug in #710. However, if you use RecBole via Pypi package or conda package, it may still not work since we haven't update our package yet.  

My suggestion is that you can download our source code from Github master branch, where the code is latest. Or you can follow the #710 and fix this bug in the package file by yourself.

**Comment by johnny12150** [comment]
Ok, I will test the newest version by cloning.
